azkaban-web-server/src/main/java/cloudflow/daos/FlowDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+package cloudflow.daos;
+
+import azkaban.db.DatabaseOperator;
+import cloudflow.error.CloudFlowNotFoundException;
+import cloudflow.models.FlowResponse;
+import cloudflow.models.Project;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FlowDaoImpl implements FlowDao {
+
+  private DatabaseOperator dbOperator;
+  private FlowAdminDaoImpl flowAdminDaoImpl;
+  private ProjectDao projectDao;
+  private static final Logger log = LoggerFactory.getLogger(ProjectDaoImpl.class);
+
+  @Inject
+  public FlowDaoImpl(DatabaseOperator dbOperator, FlowAdminDaoImpl flowAdminDaoImpl,
+      ProjectDao projectDao) {
+    this.dbOperator = dbOperator;
+    this.flowAdminDaoImpl = flowAdminDaoImpl;
+    this.projectDao = projectDao;
+  }
+
+
+  @Override
+  public Optional<FlowResponse> getFlow(String flowId){
+    List<FlowResponse> flows = new ArrayList<>();
+    FetchFlowHandler fetchFlowHandler = new FetchFlowHandler();
+    try {
+      flows = dbOperator.query(FetchFlowHandler.SELECT_FLOW_WITH_FLOW_ID,
+          fetchFlowHandler, flowId);
+      for(FlowResponse flow : flows) {
+        //flow.setAdmins(flowAdminDaoImpl.findAdminsByFlowId(flow.getId()));
+      }
+    } catch (SQLException ex) {
+      log.error("Select for flow id '{}' failed: ", flowId, ex);
+    }
+    return flows.isEmpty() ? Optional.empty() : Optional.of(flows.get(0));
+  }
+
+
+  @Override
+  public List<FlowResponse> getAllFlows(String projectId, String projectVersion) {
+    List<FlowResponse> flows = new ArrayList<FlowResponse>();
+    FetchFlowHandler fetchFlowHandler = new FetchFlowHandler();
+
+    /* If projectVersion is -1, get the latest project version */
+    if (projectVersion.equals("-1")) {
+      Optional<Project> project = projectDao.get(projectId);
+      if (!project.isPresent()) {
+        throw new CloudFlowNotFoundException("");
```

Comment:
Added error message, thanks for catching this. Changed the exception type to ProjectManagerException instead since we haven't decided which exception should be thrown when.